### PR TITLE
Heatmap: Update tooltip UX

### LINKED ISSUE #78835: Tooltip: Update Heatmap tooltip to match new design

### PR DESCRIPTION
**Pull request by adela-almasan**
Updated UI to match the new design.

heatmap-tooltip

Fixes #78835

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/panel/heatmap/tooltip/utils.ts [hunk 1]
```diff
@@ -45,7 +45,8 @@ export const formatMilliseconds = (milliseconds: number) => {
     }
   }
 
-  const unitString = value === 1 ? unit : unit + 's';
+  const plural = value !== 1 && !['ms', 's', 'm', 'h'].includes(unit);
```
Comment (leeoniya) [inline public/app/plugins/panel/heatmap/tooltip/utils.ts:48]: maybe pull this array out/above this function and do something like `const noPluralize = new Set(['ms', 's', 'm', 'h'])`. then inside the function `&& !noPluralize.has(unit)`. that way it doesnt recreate the static array on every hover. and `Set.has()` is faster than `Array.includes()` ;)

**Comment by leeoniya** [comment]
seeing some difference in Count/count/Value display in different heatmaps, not sure if we can avoid it always:

image

http://localhost:3000/d/YoacZIq7z/legacy-heatmap?orgId=1

image

feels like if we see all numeric y axis names, it should still be Bucket and not Name?

image

**Review by nmarrs — APPROVED** [review]
Noticed a potential issue with heatmap shared tooltip 😬 I think we can create a follow-up issue for this though

https://github.com/grafana/grafana/assets/22381771/aa4a6461-48fa-47f4-914d-11718e2d4cfc

Overall I think new design looks great!

old 😷 
Screenshot 2023-12-15 at 2 39 27 PM


new ❤️ 

Screenshot 2023-12-15 at 2 40 01 PM

**Comment by adela-almasan** [comment]
Follow up issues:

- https://github.com/grafana/grafana/issues/79604
- https://github.com/grafana/grafana/issues/79605 (sync is currently not functional with the new tooltips)

**Pull request merged**